services/workflows-service/src/business-report/business-report.controller.internal.ts

```diff
@@ -11,17 +11,157 @@ import {
   ListBusinessReportsDto,
   ListBusinessReportsSchema,
 } from '@/business-report/list-business-reports.dto';
-import { Prisma } from '@prisma/client';
+import { Business, BusinessReportStatus, Prisma } from '@prisma/client';
 import { ZodValidationPipe } from '@/common/pipes/zod.pipe';
+import axios from 'axios';
+import { env } from '@/env';
+import { CreateBusinessReportDto } from '@/business-report/dto/create-business-report.dto';
+import {
+  HookCallbackHandlerService,
+  ReportWithRiskScoreSchema,
+} from '@/workflow/hook-callback-handler.service';
+import { CustomerService } from '@/customer/customer.service';
+import { BusinessService } from '@/business/business.service';
+import { AlertService } from '@/alert/alert.service';
+import { Public } from '@/common/decorators/public.decorator';
+import { VerifyUnifiedApiSignatureDecorator } from '@/common/decorators/verify-unified-api-signature.decorator';
+import { BusinessReportHookBodyDto } from '@/business-report/dtos/business-report-hook-body.dto';
+import { BusinessReportHookSearchQueryParamsDto } from '@/business-report/dtos/business-report-hook-search-query-params.dto';
 
 @common.Controller('internal/business-reports')
 @swagger.ApiExcludeController()
 export class BusinessReportControllerInternal {
   constructor(
     protected readonly businessReportService: BusinessReportService,
     protected readonly logger: AppLoggerService,
+    protected readonly customerService: CustomerService,
+    protected readonly businessService: BusinessService,
+    protected readonly alertService: AlertService,
+    protected readonly hookCallbackService: HookCallbackHandlerService,
   ) {}
 
+  @common.Post()
+  @swagger.ApiOkResponse({ type: [String] })
+  @swagger.ApiForbiddenResponse({ type: errors.ForbiddenException })
+  async createBusinessReport(
+    @Body()
+    {
+      websiteUrl,
+      countryCode,
+      merchantName,
+      callbackUrl,
```

Comment:
Callback url should be generated by the server itself, 
and consist of - 

specific unique identity ( businessId / projectedId + correlationId ) 

I would think that businessId is prefered